--- vmci-only.orig/linux/vmciKernelIf.c	2015-01-17 03:06:01.000000000 +0300
+++ vmci-only/linux/vmciKernelIf.c	2015-07-04 15:16:11.994243461 +0300
@@ -1227,11 +1227,11 @@
       }
 
       if (isIovec) {
-         struct iovec *iov = (struct iovec *)src;
+         struct msghdr *msg = src;
          int err;
 
          /* The iovec will track bytesCopied internally. */
-         err = memcpy_fromiovec((uint8 *)va + pageOffset, iov, toCopy);
+         err = memcpy_from_msg((uint8 *)va + pageOffset, msg, toCopy);
          if (err != 0) {
             if (!kernelIf->isDataMapped) {
                kunmap(kernelIf->page[pageIndex]);
